### PR TITLE
Post endpoint using JSON, and real play lists

### PR DESCRIPTION
**Pull request by chaneberg**
Made play list GET endpoint return specific play lists. Added POST endpoint. Added very basic and ugly post box in order to test the POST endpoint. Improved Admin site.

Made POST endpoint use JSON, and updated client code to send it JSON. Made play lists be returned in correct format according to spec.

### CODE REVIEW

File: statler_api/views.py [hunk 1]
```diff
@@ -13,20 +15,19 @@ def healthCheck(request):
...
-    # Remove reviews from each play dict, because we aren't going
-    # that deep in the object graph
-    for aPlayDict in aPlayDictList:
-        del aPlayDict["reviews"]
-        
-    return HttpResponse(json.dumps(aPlayDictList))
+    aPlayEntryList = [vars(x) for x in aPlayEntryList]
+    for aPlayEntry in aPlayEntryList:
+        aPlayEntry["play"] = vars(aPlayEntry["play"])
+        del aPlayEntry["play"]["reviews"]
```
Comment (chaneberg) [inline statler_api/views.py:28]: @B3nD3lan3y, this is a fixed play list endpoint. The serialization is messy and will be moved into the classes by you. But the pattern of using vars() recursively might actually be useful

File: statler_api/views.py [hunk 2]
```diff
@@ -43,3 +44,20 @@ def getPlayDetail(request, play_id):
...
+def postReview(request, play_id):
+    """called when a POST request is sent to /api/play/<play_id>/reviews
+    returns json for the review added"""
+
+    # create and save a review
+    # html.escape should prevent cross-site scripting attacks
+    incomingData = json.loads(request.body.decode('utf-8'))
+    review = NewReview(play_id, html.escape(incomingData["text"]))
+    review.toDao().save()
+
```
Comment (chaneberg) [inline statler_api/views.py:58]: This is JSON serialization of the posted review.

File: statler_site/templates/play-detail-page.html [hunk 3]
```diff
@@ -50,6 +52,26 @@
...
+          dataType: 'json',
+          success: postCallback
+        });
+      }
+
+      function postCallback(response) {
+        // TODO: replace this with what we want to do when a review is posted
+        alert(JSON.stringify(response));
+      }
+
```
Comment (chaneberg) [inline statler_site/templates/play-detail-page.html:74]: @aseebart, Ben won and we're sending reviews as json. I added this script for that. You could fix the formatting of the post box and fill in the TODO above.

File: statler_site/templates/play-detail-page.html [hunk 4]
```diff
@@ -50,6 +52,26 @@
...
+      function postReview() {
+        $.ajax({
+          url: "{% url 'api:play-reviews' play_id %}",
+          type: 'post',
+          data: JSON.stringify({text: $("#post_box").val()}),
+          // Set a header containing the crsf token to prevent crsf attacks
+          // django requires this and handles it automatically
+          // see https://docs.djangoproject.com/en/1.8/ref/csrf/#ajax
+          headers: {"X-CSRFToken": "{{ csrf_token }}"},
+          dataType: 'json',
```
Comment (bencdelaney) [inline statler_site/templates/play-detail-page.html:65]: Should this be application/json?

File: statler_site/templates/play-list-page.html [hunk 5]
```diff
@@ -13,27 +15,30 @@
...
         $.getJSON(apiEndpointUrl, function(result){
           // display the returned json dynamically
-          $.each(result, function(index, value){
+          $.each(result, function(i, playEntry){
+            var play = playEntry.play;
+
+            //TODO: display plays in order by playEntry.index
 
             // this indentation helps because it looks a bit like html
             $('#play-list').append(
```
Comment (bencdelaney) [inline statler_site/templates/play-list-page.html:31]: Just as a note for the future, this should eventually be error trapped.

File: statler_api/views.py (hunk 1 above)
Comment (bencdelaney) [inline statler_api/views.py:28]: I have blown this away with my changes, and the JSON serializer automagically makes the solution recursive.

**Comment by bencdelaney** [comment]
Looks good to me. I'm cool with you merging, unless you want @aseebart to look it over first.

**Pull request merged**